### PR TITLE
Formalize data quality assurance (DQA) metrics and analytics

### PR DESCRIPTION
**Pull request by grantjllnl**
There is a need for consistent metrics to help determine the quality of data sets. The general approach to this point has been to just use a series of adhoc queries to inspect data sets and then experience to determine if the results are reasonable.

Implementation here will start with a python package encapsulating various queries. Orchestration will be done from a notebook. 

There will also be some data processing modifications along the way, starting with how PROCESS_STOP is handled.

### CODE REVIEW

File: wintappy/datautils/process_summary.sql [hunk 1]
```diff
@@ -2,8 +2,11 @@
 
 -- Create summaries for detail event types
 create or replace view process_registry_summary
+--# required
+--# template: stdview
```
Comment (lindsey-w) [inline wintappy/datautils/process_summary.sql:6]: I really like how you've added these args/options

File: wintappy/datautils/rawtostdview.sql [hunk 2]
```diff
@@ -4,47 +4,47 @@
...
 
 CREATE TABLE IF NOT EXISTS host
 AS
 SELECT
-    -- Ignore host.id.hash for now as it is really only composed of the hostname.
-    -- host.hostid.hash,
-    hostname hostname,
-    first('windows') os_family,
+    hostname,
+    list_sort(list(distinct agentid)) agent_ids,
```
Comment (lindsey-w) [inline wintappy/datautils/rawtostdview.sql:14]: can a single host have multiple agent ids?

File: wintappy/datautils/rawtostdview.sql [hunk 3]
```diff
@@ -430,27 +434,29 @@ GROUP BY ALL
 
 -- Summarize to PID_HASH+FILENAME
 
+--EXPLAIN ANALYZE
```
Comment (lindsey-w) [inline wintappy/datautils/rawtostdview.sql:437]: ?

File: wintappy/datautils/rawutil.py [hunk 4]
```diff
@@ -242,54 +318,70 @@ def create_views(con, event_map, start=None, end=None):
...
-        """
+    fqfn = resource_files(f"wintappy.schema.{sqlstmt.template}").joinpath(
+        f"{sqlstmt.name}.parquet"
+    )
+    logging.debug(f"Creating empty table from: {fqfn}")
+    sql = f"create table {sqlstmt.name} as select * from '{fqfn}' where False"
+    try:
+        con.execute(sql)
+    except CatalogException as e:
+        logging.info(
```
Comment (lindsey-w) [inline wintappy/datautils/rawutil.py:382]: should this be `logging.warning`?

**Review by lindsey-w — COMMENTED** [review]
No logic questions, looks good.

File: wintappy/datautils/rawtostdview.sql (hunk 2 above)
Comment (grantjllnl) [reply wintappy/datautils/rawtostdview.sql:14]: Yes. Likely? We don't think so.

File: wintappy/datautils/rawtostdview.sql (hunk 3 above)
Comment (grantjllnl) [reply wintappy/datautils/rawtostdview.sql:437]: Ah, debugging left over. EXPLAIN ANALYZE produces a flow-chart like output of how the query was executed by the engine and where it spent its time.

For now, I'd like to leave it in there as it's commented out. Tuning may not be over yet.

File: wintappy/datautils/rawutil.py (hunk 4 above)
Comment (grantjllnl) [reply wintappy/datautils/rawutil.py]: Yes! Fixed.

**Review by lindsey-w — APPROVED** [review]
🚢

**Pull request merged**